tests/integration/targets/lookup_sops/tasks/main.yml

```
@@ -48,6 +48,22 @@
           - "sops_lookup_simple is success"
           - "sops_success == 'foo: bar'"
 
+    - name: Test extract
+      set_fact:
+        simple_yaml_with_extract: "{{ lookup('community.sops.sops', 'extract.yaml', extract=\"['foo']\") }}"
+        simple_json_with_extract: "{{ lookup('community.sops.sops', 'extract.json', extract=\"['bar']\") }}"
+        nested_yaml_with_extract: "{{ lookup('community.sops.sops', 'extract.yaml', extract=\"['baz']['bar'][0]\") }}"
+        nested_json_with_extract: "{{ lookup('community.sops.sops', 'extract.json', extract=\"['baz']['bar'][2]\") }}"
```

This is easier to read without the escaped quotes:
```suggestion
        simple_yaml_with_extract: >-
          {{ lookup('community.sops.sops', 'extract.yaml', extract="['foo']") }}
        simple_json_with_extract: >-
          {{ lookup('community.sops.sops', 'extract.json', extract="['bar']") }}
        nested_yaml_with_extract: >-
          {{ lookup('community.sops.sops', 'extract.yaml', extract="['baz']['bar'][0]") }}
        nested_json_with_extract: >-
          {{ lookup('community.sops.sops', 'extract.json', extract="['baz']['bar'][2]") }}
```

---

Agree! This is a decent workaround for the un-pretty sops syntax.